### PR TITLE
Keymap download

### LINKED ISSUE #199: Make keymaps downloadable
This issue is needed to be implemented in order for us to crowdsource #91.

The idea is that there will be a download icon between the duplicate keymap and delete keymap icons on keymap pages. The user will be able to download the actual keymap by clicking on this icon. We could use [FileSaver.js](https://github.com/eligrey/FileSaver.js) for implementing the actual saving.

The filename should be "[KeymapName] keymap.json", so for example "QWERTY keymap.json".

The file should contain the JSON representation of the keymap inside of the following envelope:

```
{
    "site": "https://UltimateHackingKeyboard.com",
    "description": "Ultimate Hacking Keyboard keymap",
    "keyboardModel": "UHK60",
    "dataModelVersion": "{include the curent data model version here}",
    "objectType": "keymap",
    "objectValue": "{include the JSON representation of the keymap here}"
}
```

Let's not include Play Macro and Keymap Switch actions to the exported file now. We can include Play Macro actions later, but it's not needed for the time being.

### PR DESCRIPTION
**Pull request by fjozsef**
Closes #199

### CODE REVIEW

File: shared/src/components/keymap/edit/keymap-edit.component.ts [hunk 1]
```diff
@@ -41,4 +45,35 @@ export class KeymapEditComponent {
...
+            .switchMap(keymap => this.toExportableJSON(keymap))
+            .map(exportableJSON => JSON.stringify(exportableJSON));
+
+        this.keymap$
+            .combineLatest(exportableJSON$)
+            .first()
+            .subscribe(latest => {
+                const keymap = latest[0];
+                const exportableJSON = latest[1];
+                const fileName = keymap.name + '_keymap.json';
```
Comment (NejcZdovc) [inline shared/src/components/keymap/edit/keymap-edit.component.ts]: I think that name should be encoded

File: shared/src/components/keymap/edit/keymap-edit.component.ts [hunk 2]
```diff
@@ -41,4 +45,35 @@ export class KeymapEditComponent {
...
+            });
+    }
+
+    private toExportableJSON(keymap: Keymap): Observable<any> {
+        return this.store
+            .let(getUserConfiguration())
+            .first()
+            .map(userConfiguration => {
+                return {
+                    site: 'https://UltimateHackingKeyboard.com',
```
Comment (NejcZdovc) [inline shared/src/components/keymap/edit/keymap-edit.component.ts]: Site URL should be in lowercase

File: shared/src/components/keymap/header/keymap-header.component.scss [hunk 3]
```diff
@@ -43,6 +43,17 @@
     }
 }
 
+.layer__download {
+    top: 10px;
+    font-size: 0.8em;
+    position: relative;
+    margin-right: 10px;
+    &:hover {
```
Comment (NejcZdovc) [inline shared/src/components/keymap/header/keymap-header.component.scss]: Add line break above

File: shared/src/config-serializer/config-items/Keymap.ts [hunk 4]
```diff
@@ -69,4 +69,5 @@ export class Keymap {
     toString(): string {
         return `<Keymap abbreviation="${this.abbreviation}" name="${this.name}">`;
     }
+
```
Comment (NejcZdovc) [inline shared/src/config-serializer/config-items/Keymap.ts]: remove empty break

**Comment by fjozsef** [comment]
@NejcZdovc Have you run `npm install`? If you did then the postinstall script should have created the symbolic link for a shared folder in both of the electron and web folder.

File: shared/src/components/keymap/edit/keymap-edit.component.ts (hunk 1 above)
Comment (fjozsef) [inline shared/src/components/keymap/edit/keymap-edit.component.ts]: Why and how would you do that?

**Pull request merged**